src/features/profile/mySecurity/common/utils.ts

```diff
@@ -0,0 +1,302 @@
+    const res = await rpc<TPrivKeyResponse>(method, [address], config)
+    return res.result
+  } catch {
+    return null
+  }
+}
+
+async function getPastelIDs(config: TRPCConfig): Promise<TPastelID[] | null> {
+  try {
+    const res = await rpc<TPastelIDResponse>('pastelid', ['list'], config)
+    return res.result
+  } catch {
+    return null
+  }
+}
+
+export async function fetchPastelIDAndPrivateKeys(
+  config: TRPCConfig,
+): Promise<string | null> {
+  const addresses = await fetchAllAddress(config)
+  const zPrivateKeys: string[] = []
+  const tPrivateKeys: string[] = []
+
+  if (addresses) {
+    const zAddresses = addresses.zAddresses
+    for (let i = 0; i < zAddresses.length; i++) {
+      const address = zAddresses[i]
+
+      const privKey = await getPrivKeyAsString(address, 'z_exportkey', config)
+      if (privKey) {
+        zPrivateKeys.push(privKey)
+      }
+    }
+
+    const tAddresses = addresses.tAddresses
+    for (let i = 0; i < tAddresses.length; i++) {
+      const address = tAddresses[i]
+
+      const privKey = await getPrivKeyAsString(address, 'dumpprivkey', config)
+      if (privKey) {
+        tPrivateKeys.push(privKey)
+      }
+    }
+  }
+
+  const pastelIDs = await getPastelIDs(config)
+
+  if (pastelIDs?.length || zPrivateKeys.length || tPrivateKeys.length) {
+    const data = {
+      zPrivateKeys,
+      tPrivateKeys,
+      pastelIDs,
+    }
+
+    return encodeURIComponent(
+      LZUTF8.compress(JSON.stringify(data), {
+        outputEncoding: 'Base64',
+      }),
+    )
+  }
+
+  return null
+}
+
+export async function fetcAllPrivateKeys(
+  config: TRPCConfig,
+): Promise<TPrivateKey[]> {
+  const addresses = await fetchAllAddress(config)
+  const zPrivateKeys: TPrivateKey[] = []
+  const tPrivateKeys: TPrivateKey[] = []
+
+  if (addresses) {
+    const zAddresses = addresses.zAddresses
+    for (let i = 0; i < zAddresses.length; i++) {
+      const address = zAddresses[i]
+
+      const privKey = await getPrivKeyAsString(address, 'z_exportkey', config)
+      if (privKey) {
+        zPrivateKeys.push({
+          address,
+          privateKey: privKey,
+        })
+      }
+    }
+
+    const tAddresses = addresses.tAddresses
+    for (let i = 0; i < tAddresses.length; i++) {
+      const address = tAddresses[i]
+
+      const privKey = await getPrivKeyAsString(address, 'dumpprivkey', config)
+      if (privKey) {
+        tPrivateKeys.push({
+          address,
+          privateKey: privKey,
+        })
+      }
+    }
+  }
+
+  return zPrivateKeys.concat(tPrivateKeys)
+}
+
+async function importPrivKey(key: string, rescan: boolean, config: TRPCConfig) {
+  if (key.startsWith('p-secret-extended-key')) {
+    try {
+      await rpc<TPrivKeyResponse>(
+        'z_importkey',
+        [key, rescan ? 'yes' : 'no'],
+        config,
+      )
+    } catch (err) {
+      console.log(
+        `profile/mySecurity/common importPrivKey z_importkey error: ${err.message}`,
+        err,
+      )
```

Comment:
NB: consider notifying user that something went wrong, `toast` can be used for this.

So with current approach user may open a issue on github "feature doesn't work", and he even not able to see console errors with packed app. With `toast` he could see what exactly went wrong and put this info into issue

Reply:
It has been updated.